File: leetcode/java/2.AddTwoNumbers.java
// import java.lang.*;

class Node{
    int data;
    Node next;
    Node(int d){
        data = d;
        next = null;
    }
}

class Main {

    Node root;

    public void push(int data){
        Node node = new Node(data);
        if(root==null){
            root = node;
        }else{
            Node q = root;
            while(q.next != null){
                q=q.next;
            }
            q.next = node;
        }
    }

    public int getCount(){
        int count = 0;
        Node p = root;
        while(p!=null){
            count++;
            p = p.next;
        }
        return count;
    }

    public void display(){
        Node temp = root;
        while(temp!=null){
            System.out.print(temp.data + " ");
            System.out.println("");
            temp=temp.next;
        }
    }

    public int sum(int count){
        int x =0;
        Node p = root;
        if(p != null){
            for(double i=count-1;i>=0;i--){
                x += (int)(p.data * Math.pow(10,i)) ;
                p = p.next;
            }
            return x;
        }else{
            return 0;
        }

    }

    public static void main(String[] args) {
        Main l1= new Main();
        Main l2= new Main();
        l1.push(2);
        l1.push(4);
        l1.push(3);
        l2.push(5);
        l2.push(6);
        l2.push(4);
        l1.display();
        l2.display();
        int count1 = l1.getCount();
        int count2 = l2.getCount();
        int sum = l1.sum(count1) + l2.sum(count2);
        int n= sum;
        int res=0;
        while(n>1){
            res= res*10+ n%10;
            n/=10;
        }
        System.out.println("Answer: "+ res);
    }

}

File: leetcode/kotlin/2.AddTwoNumbers.kt
import java.lang.Math.pow
import kotlin.math.pow

class Node(var data:Int){
    var next:Node?=null;
}

class Main{

    var root:Node? = null;

    fun push(data: Int){
        var node= Node(data);
        if(root == null){
            root =node;
        }else{
            var q:Node? = root;
            while (q?.next != null){
                q= q?.next;
            }
            q?.next = node;
        }
    }

    fun getCount():Int{
        var count:Int =0;
        var p: Node? = root;
        while (p!=null){
            count++
            p=p.next
        }
        return count
    }

    fun sum(count:Int):Int{
        var x=0;
        var temp: Node? = root;

           for (i in count-1 downTo -1){
               x+= (temp?.data?.times(10.0.pow(i.toDouble())))?.toInt() ?: 0
               temp=temp?.next
           }
            return x

    }

    fun display(){
        var temp:Node? = root;
        while (temp != null){
            println(temp.data)
            temp = temp.next
        }
    }

}

fun main() {
    var l1= Main()
    var l2 = Main()
    l1.push(2)
    l1.push(4)
    l1.push(3)
    l2.push(5)
    l2.push(6)
    l2.push(4)
    l1.display()
    l2.display()
    var sum = l1.sum(l1.getCount())+l2.sum(l2.getCount())
    var x= sum
    var res=0
    while (x>0){
        res= res*10 + x%10
        x/=10
    }
    println("Answer : $res")
}

